### PR TITLE
Update crates. zbus 3 -> zbus 5.

### PR DESCRIPTION
**Pull request by cryptoquick**
This solves an error I was getting using this project:

https://github.com/wyhaya/tauri-plugin-theme

### CODE REVIEW

**Comment by d2weber** [comment]
Thanks for the pr! I just ran your commit locally but the example (`cargo run --example dtintanum`) seems not to work anymore for me. Did you test it and does it work for you? At first sight your code looks fine, not sure if it could be due to the changed dependencies.

**Comment by cryptoquick** [comment]
Strange. The output between main and update-crates seems identical on my end. See:

Screenshot from 2024-12-06 10-13-05

Screenshot from 2024-12-06 10-15-24

Not sure what causes that error, but I figured that was intentional.

**Comment by d2weber** [comment]
No it shouldn’t error. Unfortunately this example only works if both config files for helix and alacrity can be found, which probably isn’t the case for you. I’m sorry that this is not clearly documented. It’s like that at the moment because I use this setup on my PC. Also I should find a way to properly test interaction with the systems dbus.

You could try to comment out these lines https://github.com/d2weber/tintanum/blob/eb2a4a2a2a10415a5baa779f05e09456845d9737/examples/dtintanum/main.rs#L27-L30
I didn’t test this but then it should not anymore require any config files to be present and just print on the screen when the system dark theme changes. Similar to ‘dbus-monitor’ with the corresponding arguments. To test if it works you would have to change your systems dark mode manually and check if the running binary is reporting the changed mode.

**Comment by cryptoquick** [comment]
I see what you're saying now. I'm able to get the original to listen to theme changed events, but I'm not seeing them come across on this branch. I've poked at it a bit and I have to say I'm not really sure what the issue is. It might take some engagement on the zbus Matrix server to ask what's changed. Are you present there?

The most confounding thing is I'm not seeing errors returned from zbus, even when making calls to it.

**Comment by d2weber** [comment]
> Are you present there?

No not at the moment.

> The most confounding thing is I'm not seeing errors returned from zbus, even when making calls to it.

I just took a closer look and indeed we're getting an error when de-serializing. (I discovered this using println debugging in [`body.deserialize()?`](https://github.com/cryptoquick/tintanum/blob/cd8c950dc8949051988638d5cd6e89f6a1c51444/src/lib.rs#L43).) But I ignore errors in
https://github.com/d2weber/tintanum/blob/eb2a4a2a2a10415a5baa779f05e09456845d9737/src/lib.rs#L91
It might be reasonable to change the returned `Stream<Item = SchemePreference>` to `Result<SchemePreference>` to detect this kind of Error. In case of a correct `SchemePreference::try_from` Implementation I would expect this to error only if the system doesn't meet org.freedesktop.portal's contract though.

File: src/lib.rs [hunk 1]
```diff
@@ -39,19 +39,26 @@ impl TryFrom<zvariant::Value<'_>> for SchemePreference {
 impl<'s> TryFrom<&'s zbus::Message> for SchemePreference {
     type Error = Error;
     fn try_from(message: &'s zbus::Message) -> Result<Self> {
-        message
-            .body::<(&str, &str, zvariant::Value<'_>)>()
-            .and_then(|args| SchemePreference::try_from(args.2))
+        let body = message.body();
+        let value: zvariant::Value = body.deserialize()?;
+        SchemePreference::try_from(value)
```
Comment (d2weber) [inline src/lib.rs]:
```suggestion
        let args: (&str, &str, zvariant::Value<'_>) = body.deserialize()?;
        SchemePreference::try_from(args.2)
```

This fixes the Issue for me

**Comment by cryptoquick** [comment]
Just tested it, it works great now!

**Review by d2weber — APPROVED** [review]

**Comment by d2weber** [comment]
> Just tested it, it works great now!

Cool! Thank you very much!

**Pull request merged**